rclcpp/include/rclcpp/node.hpp

```diff
@@ -303,6 +309,24 @@ class Node : public std::enable_shared_from_this<Node>
   bool
   get_parameter(const std::string & name, ParameterT & parameter) const;
 
+  /// Assign the value of the map parameter if set into the values argument.
+  /**
+   * Parameter names that are part of a map are of the form "name.member".
```

Comment:
While that is true (and makes some sense for `set`), it doesn't really make much sense for `get`.  If we just make it so that there is no "grouping" with this API, then there is little reason to add this API; we can do the same thing by calling `get_parameters` with a vector of names.  The explicit purpose of the map-based one is to have a group of variables that we can get into a single map.  This is similar to the map-based [getParam](https://github.com/ros/ros_comm/blob/melodic-devel/clients/roscpp/include/ros/node_handle.h#L1806) overload in ROS1.